test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -827,6 +832,24 @@ void restart(RestartCallback callback) throws Exception {
                     IOUtils.rm(nodeEnv.nodeDataPaths());
                 }
             }
+            startNewNode(newSettings);
+        }
+
+        void stopNodeWithoutClearingData() throws IOException {
```

Comment:
Not sure if this is the best way to achieve what I wanted - which was stopping a node and subsequently restarting the node with the same data directories, while executing other tasks in between.  I didn't see anything in the `InternalTestCluster` that provided that, so I added this functionality.